packages/flutter/lib/src/material/page_transitions_theme.dart

```diff
@@ -177,132 +176,188 @@ class _ZoomPageTransition extends StatefulWidget {
+          BuildContext context,
+          Animation<double> animation,
+          Widget child,
+        ) {
+          return _ZoomEnterTransition(
+            animation: animation,
+            reverse: true,
+            child: child,
+          );
+        },
+        reverseBuilder: (
+          BuildContext context,
+          Animation<double> animation,
+          Widget child,
+        ) {
+          return _ZoomExitTransition(
+            animation: animation,
+            child: child,
+          );
+        },
+        child: child,
+      ),
+    );
+  }
 }
 
-class __ZoomPageTransitionState extends State<_ZoomPageTransition> {
-  AnimationStatus _currentAnimationStatus;
-  AnimationStatus _lastAnimationStatus;
+class _ZoomEnterTransition extends StatelessWidget {
+  const _ZoomEnterTransition({
+    Key key,
+    this.animation,
+    this.reverse = false,
+    this.child,
+  }) : super(key: key);
 
-  @override
-  void initState() {
-    super.initState();
-    widget.animation.addStatusListener((AnimationStatus animationStatus) {
-      _lastAnimationStatus = _currentAnimationStatus;
-      _currentAnimationStatus = animationStatus;
-    });
-  }
+  final Animation<double> animation;
+  final Widget child;
+  final bool reverse;
 
-  // This check ensures that the animation reverses the original animation if
-  // the transition were interruped midway. This prevents a disjointed
-  // experience since the reverse animation uses different fade and scaling
-  // curves.
-  bool get _transitionWasInterrupted {
-    bool wasInProgress = false;
-    bool isInProgress = false;
-
-    switch (_currentAnimationStatus) {
-      case AnimationStatus.completed:
-      case AnimationStatus.dismissed:
-        isInProgress = false;
-        break;
-      case AnimationStatus.forward:
-      case AnimationStatus.reverse:
-        isInProgress = true;
-        break;
-    }
-    switch (_lastAnimationStatus) {
-      case AnimationStatus.completed:
-      case AnimationStatus.dismissed:
-        wasInProgress = false;
-        break;
-      case AnimationStatus.forward:
-      case AnimationStatus.reverse:
-        wasInProgress = true;
-        break;
+  static final Animatable<double> _fadeInTransition = Tween<double>(
+    begin: 0.0,
+    end: 1.00,
+  ).chain(CurveTween(curve: const Interval(0.125, 0.250)));
+
+  static final Animatable<double> _scaleDownTransition = Tween<double>(
+    begin: 1.10,
+    end: 1.00,
+  ).chain(_ZoomPageTransition._scaleCurveSequence);
+
+  static final Animatable<double> _scaleUpTransition = Tween<double>(
+    begin: 0.85,
+    end: 1.00,
+  ).chain(_ZoomPageTransition._scaleCurveSequence);
+
+  static final Animatable<double> _scrimOpacityTween = Tween<double>(
+    begin: 0.0,
+    end: 0.60,
+  ).chain(CurveTween(curve: const Interval(0.2075, 0.4175)));
+
+  @override
+  Widget build(BuildContext context) {
+    double opacity = 0;
+    // Only modify the scrim opacity when running forwards and
```

Comment:
A comment should explain why code is written the way it is, it should not paraphrase the code in words like this one is doing. In particular, this comment should explain why the animation status matters (this is still not clear to me).

Reply:
It's really hard for a person whose native language is not English 😕. I'm wondering if this looks okay:
```dart
// When the transition is running with the [reverseBuilder] in [DualTransitionBuilder],
// the status of the animation will be always [AnimationStatus.completed].
//
// To ensure the opacity only changed with forward animation, we need to combine
// the [reverse] flag with the animation status.
```

Reply:
@goderbauer